### ISSUE TITLE
[gatsby-source-contentful] Filtering by locale simply uses fallback values

### ISSUE THREAD

**Issue by jessepinho** [gatsbyjs/gatsby#14077]
## Description

I can't figure out how to make a GraphQL query for articles in just a particular locale, because the fallback behavior seems to result in nodes being created for locales even though those nodes shouldn't exist yet.

Let's say I want to get all Portuguese articles:
```
{
  allContentfulArticle(filter: {node_locale: {eq: "pt"}, slug: {regex: "/.+/"}}) {
    edges {
      node {
        node_locale
        slug
      }
    }
  }
}
```

This query returns all my articles, _including_ English ones. But the `node_locale` value is still set to `pt` in every one. I think it's because the slug is falling back to English.

In short, I'm trying to get _only entries that have a slug in Portuguese_. Instead, I'm getting literally nonexistent nodes, filled in with the English values.

### Steps to reproduce

*Note: I've created a minimal reproduction in [CodeSandbox](https://codesandbox.io/embed/l7kpnnjwk7). If you want to make GraphQL queries in the sandbox to see what I'm describing, navigate to `/___graphql` in the CodeSandbox browser. You can explore the Contentful schema and data at [discovery.contentful.com](https://discovery.contentful.com/entries/by-content-type?delivery_access_token=5SVOC9G5NAvmxvrA2k4kEi-NrB5OK8OpMFB5LYApdqk&preview=false&preview_access_token=&space_id=hhasyt0xx1l5).*

1. So, let's say I have a new Contentful space for which I've imported some English articles via the UI. I add Portuguese as a locale, with a fallback to English.
1. I make a GraphQL query for _Portuguese-only_ articles, querying for the slug field. It returns 0 nodes, as expected, since none of the articles have fields filled in for Portuguese.
1. Then, I go to Contentful, click one of the English articles, and _without editing any fields at all_, I click "Publish."
1. Then, I go back to GraphiQL and make the same query.

### Expected result
Just as before, there should be 0 nodes in Portuguese.

### Actual result
There is now 1 node — the one I just clicked "Save" on — and its slug is in English.

*Conclusion:*
1. Contentful must be creating a Portuguese node when I click "Save" on an English article, even if I haven't filled in any Portuguese fields. 
1. gatsby-source-contentful doesn't realize these nodes are empty, and treats them as though they exist, and simply falls back to the default locale.

### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here.

```

  System:
    OS: macOS 10.14.4
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 11.12.0 - /usr/local/bin/node
    Yarn: 1.5.1 - node_modules/.bin/yarn
    npm: 6.7.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/local/bin/python
  Browsers:
    Chrome: 74.0.3729.157
    Firefox: 66.0.2
    Safari: 12.1
  npmPackages:
    gatsby: ^2.1.19 => 2.3.14 
    gatsby-image: ^2.0.30 => 2.0.37 
    gatsby-plugin-react-helmet: ^3.0.7 => 3.0.12 
    gatsby-plugin-sharp: ^2.0.23 => 2.0.32 
    gatsby-plugin-sitemap: ^2.1.0 => 2.1.0 
    gatsby-plugin-styled-components: ^3.0.6 => 3.0.7 
    gatsby-plugin-typescript: ^2.0.9 => 2.0.12 
    gatsby-source-contentful: ^2.0.33 => 2.0.45 
    gatsby-transformer-remark: ^2.3.0 => 2.3.8 
    gatsby-transformer-sharp: ^2.1.15 => 2.1.17 
```

**Comment by jessepinho** [comment]
By the way, this issue was raised in #4531 by @anjawaldbauer. I believe it's the same issue, actually.

**Comment by pieh** [comment]
This is pretty difficult issue, because of how Contentful handles localization. The localization is not entry based - but it's field based.

That means when you use locale with fallback, and translate just title (but not body). You will get partially translated entry - with `pt` title, but `en` body (fallback) - what locale should be reported for this entry?

**Comment by jessepinho** [comment]
@pieh Note that I included a regex for the slug field to ensure that the slug field _specifically_ is filled in in the given locale. That's why I expected it to return only entries with the Portuguese locale.

Also, note that the Contentful API supports what I'm trying to do! See, for example, the API call I'm making for [this page](https://helloclue.com/pt/artigos/ciclo-a-z) on helloclue.com for Portuguese articles: `https://cdn.contentful.com/spaces/juauvlea4rbf/environments/master/entries?content_type=featuredImage&fields.slug%5Bexists%5D=true&fields.category.sys.contentType.sys.id=category&fields.category.fields.slug=ciclo-a-z&locale=pt&limit=1000&order=-fields.publishDate` (Ignore the `content_type=featuredImage` — that's a legacy content type name that actually refers to our article content type.) Note that I set `fields.slug[exists]=true`, and as a result, it only returns articles with a Portuguese slug filled in. 

Thus, the behavior of `gatsby-source-contentful` doesn't match that of the Contentful API, right? Am I missing something?
